connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -84,13 +69,24 @@ final case class InvalidCommandInput(
     private val cause: Throwable = null)
     extends Exception(message, cause)
 
-class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
+class SparkConnectPlanner(
+    val sessionHolder: SessionHolder,
+    responseHandlerOpt: Option[SparkConnectResponseHandler[_]] = None)
```

Comment:
Is there a valid use-case where this can be empty? Personally, I don't think that this should ever be none. I understand for testing purposes one might inject a mock, but otherwise the planner cannot exist without the responsehandler

Reply:
There exists some test case only need transform proto to `LogicalPlan` and doesn't depend on `responsehandler`.

Please refer: https://github.com/apache/spark/blob/1a5b9f295a623901100b6f41e785138fba0bc156/connector/connect/server/src/test/scala/org/apache/spark/sql/connect/ProtoToParsedPlanTestSuite.scala#L181